src/test/java/org/apache/bcel/generic/CountVisitor.java

```diff
@@ -0,0 +1,2012 @@
+
+    @Override
+    public void visitGETFIELD(GETFIELD obj) {
+        visitGETFIELDCount++;
+    }
+
+    private int visitGETSTATICCount = 0;
+
+    @Override
+    public void visitGETSTATIC(GETSTATIC obj) {
+        visitGETSTATICCount++;
+    }
+
+    private int visitGOTOCount = 0;
+
+    @Override
+    public void visitGOTO(GOTO obj) {
+        visitGOTOCount++;
+    }
+
+    private int visitGOTO_WCount = 0;
+
+    @Override
+    public void visitGOTO_W(GOTO_W obj) {
+        visitGOTO_WCount++;
+    }
+
+    private int visitGotoInstructionCount = 0;
+
+    @Override
+    public void visitGotoInstruction(GotoInstruction obj) {
+        visitGotoInstructionCount++;
+    }
+
+    private int visitI2BCount = 0;
+
+    @Override
+    public void visitI2B(I2B obj) {
+        visitI2BCount++;
+    }
+
+    private int visitI2CCount = 0;
+
+    @Override
+    public void visitI2C(I2C obj) {
+        visitI2CCount++;
+    }
+
+    private int visitI2DCount = 0;
+
+    @Override
+    public void visitI2D(I2D obj) {
+        visitI2DCount++;
+    }
+
+    private int visitI2FCount = 0;
+
+    @Override
+    public void visitI2F(I2F obj) {
+        visitI2FCount++;
+    }
+
+    private int visitI2LCount = 0;
+
+    @Override
+    public void visitI2L(I2L obj) {
+        visitI2LCount++;
+    }
+
+    private int visitI2SCount = 0;
+
+    @Override
+    public void visitI2S(I2S obj) {
+        visitI2SCount++;
+    }
+
+    private int visitIADDCount = 0;
+
+    @Override
+    public void visitIADD(IADD obj) {
+        visitIADDCount++;
+    }
+
+    private int visitIALOADCount = 0;
+
+    @Override
+    public void visitIALOAD(IALOAD obj) {
+        visitIALOADCount++;
+    }
+
+    private int visitIANDCount = 0;
+
+    @Override
+    public void visitIAND(IAND obj) {
+        visitIANDCount++;
+    }
+
+    private int visitIASTORECount = 0;
+
+    @Override
+    public void visitIASTORE(IASTORE obj) {
+        visitIASTORECount++;
+    }
+
+    private int visitICONSTCount = 0;
+
+    @Override
+    public void visitICONST(ICONST obj) {
+        visitICONSTCount++;
+    }
+
+    private int visitIDIVCount = 0;
+
+    @Override
+    public void visitIDIV(IDIV obj) {
+        visitIDIVCount++;
+    }
+
+    private int visitIF_ACMPEQCount = 0;
+
+    @Override
+    public void visitIF_ACMPEQ(IF_ACMPEQ obj) {
+        visitIF_ACMPEQCount++;
+    }
+
+    private int visitIF_ACMPNECount = 0;
+
+    @Override
+    public void visitIF_ACMPNE(IF_ACMPNE obj) {
+        visitIF_ACMPNECount++;
+    }
+
+    private int visitIF_ICMPEQCount = 0;
+
+    @Override
+    public void visitIF_ICMPEQ(IF_ICMPEQ obj) {
+        visitIF_ICMPEQCount++;
+    }
+
+    private int visitIF_ICMPGECount = 0;
+
+    @Override
+    public void visitIF_ICMPGE(IF_ICMPGE obj) {
+        visitIF_ICMPGECount++;
+    }
+
+    private int visitIF_ICMPGTCount = 0;
+
+    @Override
+    public void visitIF_ICMPGT(IF_ICMPGT obj) {
+        visitIF_ICMPGTCount++;
```

Comment:
Don't use underscores.

Reply:
Any suggestion on how to name `visitIF_ICMPGTCount`?
I just use the convention `visitXXXCount` for all count variables.

Reply:
This a _visitor_ that _only_ counts, so naming every ivars `visitSomethingCount` feels redundant to me, I would keep the names as simple as possible like `something`.

Reply:
Updated the naming as suggested. Tried my best to follow variable naming convention. But there are some special case like the class name lower case is java keyword "RETURN", "NEW", "GOTO", leading to variable name like "rETURN", "nEW" and "goTo"